jvector-base/src/main/java/io/github/jbellis/jvector/pq/KMeansPlusPlusClusterer.java

```diff
@@ -143,18 +143,24 @@ private float[][] chooseInitialCentroids(float[][] points) {
 
     /**
      * Assigns points to the nearest cluster.  The results are stored as ordinals in `assignments`
+     *
+     * @param firstRun whether this is the first assignment of points, which means the previous assignment is meaningless
+     * @return the number of points that changed clusters
      */
-    private int assignPointsToClusters() {
+    private int assignPointsToClusters(boolean firstRun) {
         int changedCount = 0;
 
         for (int i = 0; i < points.length; i++) {
             float[] point = points[i];
-            var oldAssignment = assignments[i];
+            var oldAssignment = firstRun ? -1 : assignments[i];
```

Comment:
could we get rid of the firstRun flag and the if statements by just calling Arrays.fill(assignments, -1) to start with?